src/stores/WalletStore.ts

```diff
@@ -19,42 +24,76 @@ export default class WalletStore {
   private static GET_INFO_INTERVAL_MS: number = 30000;
   private static GET_PRICE_INTERVAL_MS: number = 60000;
 
-  @observable public loading = true;
+  @observable public loading = INIT_VALUES.loading;
+  @observable public appSalt?: Uint8Array = INIT_VALUES.appSalt;
+  @observable public passwordHash?: string = INIT_VALUES.passwordHash;
   @observable public info?: Insight.IGetInfo = INIT_VALUES.info;
-  @observable public accounts: Account[] = INIT_VALUES.accounts;
+  @observable public testnetAccounts: Account[] = INIT_VALUES.testnetAccounts;
+  @observable public mainnetAccounts: Account[] = INIT_VALUES.mainnetAccounts;
   @observable public loggedInAccount?: Account = INIT_VALUES.loggedInAccount;
   @observable public qtumPriceUSD = 0;
-
-  @computed public get balanceUSD() {
+  @computed public get accounts(): Account[] {
+    return this.app.networkStore.isMainNet ? this.mainnetAccounts : this.testnetAccounts;
+  }
+  @computed public get balanceUSD(): string {
     if (this.qtumPriceUSD && this.info) {
       return (this.qtumPriceUSD * this.info.balance).toFixed(2);
     } else {
       return 'Loading';
     }
   }
-
   public wallet?: Wallet = INIT_VALUES.wallet;
 
   private app: AppStore;
-  private getInfoInterval?: number = undefined;
-  private getPriceInterval?: number = undefined;
+  private getInfoInterval?: NodeJS.Timer = undefined;
+  private getPriceInterval?: NodeJS.Timer = undefined;
 
   constructor(app: AppStore) {
     this.app = app;
-    this.getAccountsFromStorage();
+    this.fetchStorageValues();
   }
 
-  public getAccountsFromStorage() {
-    if (this.app.networkStore.isMainNet) {
-      // Set the existing accounts from Chrome storage
-      chrome.storage.local.get(STORAGE.MAINNET_ACCOUNTS, ({ mainnetAccounts }) => {
-        this.setAccountsAndRoute(mainnetAccounts);
-      });
+  /*
+  * Creates a new passwordHash or validates the existing one for the main login.
+  * @param password {string} The new/existing password for the per-install appSalt.
+  */
+  @action
+  public login = async (password: string) => {
+    this.loading = true;
+
+    // TODO: make this async so it doesnt block the UI
+    // Generate appSalt if needed
+    if (!this.appSalt) {
+      const appSalt: Uint8Array = window.crypto.getRandomValues(new Uint8Array(16)) as Uint8Array;
+      this.appSalt = appSalt;
+      chrome.storage.local.set(
+        { [STORAGE.APP_SALT]: appSalt.toString() },
+        () => console.log('appSalt set'),
+      );
+    }
+
+    // Derive passwordHash
+    const saltBuffer = Buffer.from(this.appSalt!);
```

Comment:
maybe create a getter for `passwordHash`?

Reply:
password hash should throw error if appSalt is empty

Reply:
done